src/javascript/app_2/Stores/client_store.js

```diff
@@ -1,12 +1,366 @@
-import { observable } from 'mobx';
-import Client         from '_common/base/client_base';
-import BaseStore      from './base_store';
+import {
+    action,
+    computed,
+    observable,
+    when }                     from 'mobx';
+import moment                  from 'moment';
+import {
+    requestLogout,
+    WS }                       from 'Services';
+import { getAccountTitle }     from '_common/base/client_base';
+import GTM                     from '_common/base/gtm';
+import BinarySocket            from '_common/base/socket_base';
+import * as SocketCache        from '_common/base/socket_cache';
+import { localize }            from '_common/localize';
+import {
+    LocalStore,
+    State }                    from '_common/storage';
+import BaseStore               from './base_store';
+import { buildCurrenciesList } from './Modules/Trading/Helpers/currency';
 
+const storage_key = 'client.accounts';
 export default class ClientStore extends BaseStore {
-    @observable balance;
-    @observable is_logged_in = !!Client.isLoggedIn();
-    @observable loginid      = Client.get('loginid');
-    @observable currency     = Client.get('currency') || '';
-    @observable upgrade_info = Client.getBasicUpgradeInfo();
-    @observable can_upgrade  = this.upgrade_info.can_upgrade || this.upgrade_info.can_open_multi;
+    @observable loginid;
+    @observable upgrade_info;
+    @observable accounts;
+    @observable switched         = '';
+    @observable switch_broadcast = false;
+    @observable currencies_list  = {};
+    @observable selected_currency = '';
+
+    constructor(root_store) {
+        super({ root_store });
+    }
+
+    @computed
+    get balance() {
+        if (!this.accounts) return '';
+        return (this.accounts[this.loginid] && this.accounts[this.loginid].balance) ?
+            this.accounts[this.loginid].balance.toString() :
+            '';
+    }
+
+    /**
+     * Temporary property. should be removed once we are fully migrated from the old app.
+     *
+     * @returns {boolean}
+     */
+    @computed
+    get is_client_allowed_to_visit() {
+        return !!(
+            !this.is_logged_in || this.is_virtual ||
+            this.accounts[this.loginid].landing_company_shortcode === 'costarica'
+        );
+    }
+
+    @computed
+    get account_list() {
+        return this.all_loginids.map(id => (
+            id !== this.loginid &&
+            !this.isDisabled(id) &&
+            this.getToken(id) ?
+                this.getAccountInfo(id) :
+                undefined
+        )).filter(account => account);
```

Comment:
i think we can remove this filter since it does nothing ?

Reply:
it removes falsies.